### PR TITLE
Port endpoints.go test to new-style Gingko test

### PR DESCRIPTION
**Pull request by zmerlynn**

### CODE REVIEW

**Comment by zmerlynn** [comment]
cc @satnam6502 @filbranden @brendanburns @rrati

**Comment by satnam6502** [comment]
I will wait a bit before merging in case there are any comments.

**Comment by filbranden** [comment]
Looks good to me.

File: test/e2e/service.go [hunk 1]
```diff
@@ -180,4 +181,131 @@ var _ = Describe("Services", func() {
...
+	It("should serve basic a endpoint from pods", func(done Done) {
+		serviceName := "endpoint-test"
+		ns := api.NamespaceDefault
+		labels := map[string]string{
+			"foo": "bar",
+			"baz": "blah",
+		}
+
+		defer func() {
+			err := c.Services(ns).Delete(serviceName)
```
Comment (roberthbailey) [inline test/e2e/service.go]: Do we need a GinkgoRecover() call in each deferred func?

Comment (filbranden) [inline test/e2e/service.go]: According to the docs, it's only needed in a goroutine.

http://onsi.github.io/ginkgo/#marking-specs-as-failed

Does a `defer` block constitue a goroutine? I thought it was run in the main thread... I think it should be possible to write an experiment with a failure inside the `defer` block to see whether Ginkgo is actually catching the error...

Comment (zmerlynn) [inline test/e2e/service.go]: Yeah, @roberthbailey and I just chatted and that was my understanding as well. I believe the defer's run at the tail of the main goroutine here. We should test this case eventually, though ...

Comment (roberthbailey) [inline test/e2e/service.go]: I must have been mis-remembering where I saw it in previous test code.

**Pull request merged**